src/x86_parser.rs

```diff
@@ -215,16 +219,34 @@ pub fn populate_instructions(xml_contents: &str) -> anyhow::Result<Vec<Instructi
     }
 
     // provide a URL example page -----------------------------------------------------------------
-    // parse this x86 page, grab the contents of the table + the URLs they are referring to
-    // TODO Fetching fliecloutier.com may take time and sometimes it's unresponsive. Cache this
-    //      once and re-use.
-    // TODO Add a CLI argument to refresh this cache.
+    // 1. If the cache refresh option is enabled or the cache doesn't exist, attempt to fetch the
+    //    data, write it to the cache, and then use it
+    //      - parse this x86 page, grab the contents of the table + the URLs they are referring to
+    // 2. Otherwise, attempt to read the data from the cache
+    let cache_refresh = args() // replace with an actual way to check CLI args
+        .any(|arg| arg.contains("refresh"));
```

Comment:
Could we make this look for `--refresh-cache` instead?

Also, I'm seeing the `// replace with an actual way to check CLI args` comment.
Is there anything additional to do here? Do you want to use a CLI parser instead ?

Reply:
I don't have anything already implemented re this. We can do this in the future though. :)